### PR TITLE
UPSTREAM: add imports for map conversion types

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
[test]

@smarterclayton ptal.

**Comment by bparees** [comment]
Also I marked this "UPSTREAM" but there's no equivalent upstream commit yet, mostly because there doesn't appear to be an equivalent upstream commit for the work it's based on.  I'm probably confused on our processes again.

**Comment by smarterclayton** [comment]
It's 9971

On Jul 5, 2015, at 3:40 PM, Ben Parees <notifications@github.com> wrote:

Also I marked this "UPSTREAM" but there's no equivalent upstream commit
yet, mostly because there doesn't appear to be an equivalent upstream
commit for the work it's based on. I'm probably confused on our processes
again.

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/3567#issuecomment-118660577>.

**Comment by bparees** [comment]
ok.  assuming you're ok w/ my fix, do you just want to add it into your existing PR against upstream then?

**Comment by smarterclayton** [comment]
Yes, will look at fix later

On Jul 5, 2015, at 3:43 PM, Ben Parees <notifications@github.com> wrote:

ok. assuming you're ok w/ my fix, do you just want to add it into your
existing PR against upstream then?

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/3567#issuecomment-118660802>.

**Comment by bparees** [comment]
@smarterclayton is there someone else who can sign off on this while you're out?  it's blocking my api update PR which is getting merge conflicted by pretty much everyone, so i'd like to get it in.

@deads2k @derekwaynecarr ?

**Comment by liggitt** [comment]
Add the upstream PR number into your commit message (e.g. `UPSTREAM: <number>: Commit message`) for future rebasers

**Comment by liggitt** [comment]
out of curiosity, do the conversions need to be regenerated after this, or is this blocking something else in progress?

**Comment by bparees** [comment]
yes the conversions have to be regenerated after this and the ones that get generated don't build without this fix.

**Comment by bparees** [comment]
i've added a pointer to clayton's PR to my commit, but his PR 
1) isn't merged yet
2) doesn't contain this change yet

so...yeah.

**Comment by deads2k** [comment]
@bparees open a pull against @smarterclayton's branch upstream and point to it from here so that there's some reasonable chance this will be findable in the next rebase.  If this is missing, we'll get a compile break, correct?

File: Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/runtime/conversion_generator.go [hunk 1]
```diff
@@ -185,11 +185,15 @@ func isComplexType(reflection reflect.Type) bool {
 func (g *conversionGenerator) generateConversionsForMap(inType, outType reflect.Type) error {
 	inKey := inType.Key()
 	outKey := outType.Key()
+	g.addImportByPath(inKey.PkgPath())
```
Comment (deads2k) [inline Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/runtime/conversion_generator.go]: I'm not very familiar with the generators, but I would have expected a corresponding deep copy change.  Why isn't it needed?

Comment (deads2k) [inline Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/runtime/conversion_generator.go]: This already covers it? https://github.com/openshift/origin/blob/master/Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/runtime/deep_copy_generator.go#L154

**Comment by deads2k** [comment]
Question about conversion versus deep_copy generators and a note to be sure that pull to the upstream branch is made.  Then lgtm.

File: Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/runtime/conversion_generator.go (hunk 1 above)
Comment (bparees) [inline Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/runtime/conversion_generator.go]: looks that way.

**Comment by bparees** [comment]
link to the pull to @smarterclayton's branch that adds this change:
https://github.com/smarterclayton/kubernetes/pull/4

**Comment by deads2k** [comment]
lgtm [merge]

**Pull request merged**